parser-native/src/main/java/ru/urururu/sanity/cpp/NativeInstructionParser.java

```diff
@@ -0,0 +1,478 @@
+                    bitreader.LLVMGetOperand(instruction, 2), bitreader.LLVMGetOperand(instruction, 1));
+        }
+    }
+
+    private class SwitchParser extends AbstractParser {
+        @Override
+        public LLVMOpcode getOpcode() {
+            return LLVMOpcode.LLVMSwitch;
+        }
+
+        @Override
+        public Cfe parse(NativeCfgBuildingCtx ctx, SWIGTYPE_p_LLVMOpaqueValue instruction) {
+            List<SWIGTYPE_p_LLVMOpaqueValue> values = new ArrayList<>();
+            List<SWIGTYPE_p_LLVMOpaqueValue> labels = new ArrayList<>();
+
+            int i = 2;
+            while (i + 1 < bitreader.LLVMGetNumOperands(instruction)) {
+                values.add(bitreader.LLVMGetOperand(instruction, i++));
+                labels.add(bitreader.LLVMGetOperand(instruction, i++));
+            }
+
+            return createSwitch(ctx, instruction, bitreader.LLVMGetOperand(instruction, 0), bitreader.LLVMGetOperand(instruction, 1), values, labels);
+        }
+    }
+
+    private class BinaryOperationParser extends AbstractParser {
+        private final Map<LLVMOpcode, BinaryExpression.Operator> opcodeOperatorMap;
+
+        BinaryOperationParser() {
+            opcodeOperatorMap = FinalMap.createHashMap();
+            opcodeOperatorMap.put(LLVMOpcode.LLVMAdd, BinaryExpression.Operator.Add);
+            opcodeOperatorMap.put(LLVMOpcode.LLVMFAdd, BinaryExpression.Operator.Add);
+            opcodeOperatorMap.put(LLVMOpcode.LLVMSub, BinaryExpression.Operator.Sub);
+            opcodeOperatorMap.put(LLVMOpcode.LLVMFSub, BinaryExpression.Operator.Sub);
+            opcodeOperatorMap.put(LLVMOpcode.LLVMMul, BinaryExpression.Operator.Mul);
+            opcodeOperatorMap.put(LLVMOpcode.LLVMFMul, BinaryExpression.Operator.Mul);
+            opcodeOperatorMap.put(LLVMOpcode.LLVMUDiv, BinaryExpression.Operator.Div);
+            opcodeOperatorMap.put(LLVMOpcode.LLVMSDiv, BinaryExpression.Operator.Div);
+            opcodeOperatorMap.put(LLVMOpcode.LLVMFDiv, BinaryExpression.Operator.Div);
+            opcodeOperatorMap.put(LLVMOpcode.LLVMURem, BinaryExpression.Operator.Rem);
+            opcodeOperatorMap.put(LLVMOpcode.LLVMSRem, BinaryExpression.Operator.Rem);
+            opcodeOperatorMap.put(LLVMOpcode.LLVMFRem, BinaryExpression.Operator.Rem);
+
+            opcodeOperatorMap.put(LLVMOpcode.LLVMAnd, BinaryExpression.Operator.And);
+            opcodeOperatorMap.put(LLVMOpcode.LLVMOr, BinaryExpression.Operator.Or);
+            opcodeOperatorMap.put(LLVMOpcode.LLVMXor, BinaryExpression.Operator.Xor);
+
+            opcodeOperatorMap.put(LLVMOpcode.LLVMShl, BinaryExpression.Operator.ShiftLeft);
+            opcodeOperatorMap.put(LLVMOpcode.LLVMLShr, BinaryExpression.Operator.ShiftRight);
+            opcodeOperatorMap.put(LLVMOpcode.LLVMAShr, BinaryExpression.Operator.ShiftRight);
+        }
+
+        @Deprecated
+        public BinaryOperationParser(LLVMOpcode opcode, BinaryExpression.Operator operator) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Add the missing @deprecated Javadoc tag. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1133)